spec.md

```diff
@@ -1450,6 +1464,44 @@ message NodeServiceCapability {
 
 If the plugin is unable to complete the NodeGetCapabilities call successfully, it MUST return a non-ok gRPC code in the gRPC status.
 
+
+#### `NodeGetInfo`
+
+A Node Plugin MUST implement this RPC call if the plugin has `PUBLISH_UNPUBLISH_VOLUME` controller capability.
+The Plugin SHALL assume that this RPC will be executed on the node where the volume will be used.
+The CO SHOULD call this RPC for the node at which it wants to place the workload.
+The result of this call will be used by CO in `ControllerPublishVolume`.
+
+```protobuf
+message NodeGetInfoRequest {
+}
+
+message NodeGetInfoResponse {
```

Comment:
This change doesn't allow nodes to return any more opaque data to the controller than the old spec. It's still just a single string (node_id). An opaque string map would give more flexibility to implementers.

Reply:
Yeah it was discussed that, `NodeGetInfo` will be extended to cover any immutable/invariant data about the node. So this field has to be extended. We are going to need a follow up PR because this PR just concerns itself with node volume limits.

Reply:
some history: I think the issue w/ returning a `map` or `repeated` for `node_id` was that it doesn't hash well: implementations want to use the `node_id` as a dictionary key, which is more difficult unless `node_id` has a primitive type.